pkg/scheduler/ugm/queue_tracker.go

```diff
@@ -94,67 +97,33 @@ func (qt *QueueTracker) increaseTrackedResource(hierarchy []string, applicationI
 	}
 	finalResourceUsage := qt.resourceUsage.Clone()
 	finalResourceUsage.AddTo(usage)
-	wildCardQuotaExceeded := false
 	existingApp := qt.runningApplications[applicationID]
 
 	// apply user/group specific limit settings set if configured, otherwise use wild card limit settings
 	if qt.maxRunningApps != 0 && !resources.IsZero(qt.maxResources) {
-		log.Log(log.SchedUGM).Debug("applying enforcement checks using limit settings of specific user/group",
+		log.Log(log.SchedUGM).Debug("applying enforcement checks using limit settings",
```

Comment:
I might be wrong, wasn't getting rid of this whole thing the main idea? As far as I can remember, we agreed that this check here is nothing more than an internal state check at this point, because maxrunning apps & resource limits are already verified at this point by `CanRunApp()` & `Headroom()` so technically (if the impl is 100% correct) we should not enter this branch.

Reply:
We do many things between `CanRunApp()/Headroom()` and `Increase` as part of scheduling cycle. So, things could have changed in the middle, which may cause issues to one of the two different go routines crossed earlier check simultaneously.

Reply:
"So, things could have changed in the middle" - how exactly? That would mean that invariants can be violated. I was under the impression that certain things only change from the scheduling cycle. So we deny an app to run (from `canRunApp()`) and for some reason the number of running apps still goes above the limit?